x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorConstants.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+public class ConnectorConstants {
```

Comment:
I think constant classes is an anti-pattern. Could each constant be defined in the classes that makes more sense to define them?

Reply:
I'd be happy to move the constants, was just copying the pattern from Behavioral Analytics here.

Reply:
My concerns about constant classes (we can continue offline, but as I opened my mouth, let's add some reasoning) 😉 :
- Constants are no longer logically related to the class that should define them
- Makes navigation harder (want to know what value that constant has? Go somewhere else)
- Can become cumbersome to maintain (40+ constants in a single class)
- Creates an artificial dependency for other classes (everyone references the Constant class)

#1 is the top reason for me - I find classes  should contain everything that is logically related to them.

I only see this need in case of math calculations or the like, where there are constants that are not related to specific classes, or constants that are used extensively in a specific module. In that case, creating a Utility class (private constructor, final class) would be the way to go.

Happy to discuss offline!